docs/README.md

```diff
@@ -119,19 +118,58 @@ options:
                         Sets the MFA response to a challenge
   --quiet               Suppress output```
 ```
+Regarding the Okta password, we are fans of automation but do not
+recommend passing in the password to tokendito via plaintext or storing
+it in your environment locally.
 
-## Supported MFA Options
 
--   Native Okta factors (push, call, sms, TOTP) except Biometrics (FIDO
-    webauthn)
+# Environment variables and user configuration
+tokendito supports the use of environment variables and user configuration equivalent to specify the default values for most options.
+
+## Precedence
+Credentials and configuration settings take precedence in the following order:  
+1) Command line options -- Overrides settings in any other location. You can specify \--username, \--role-arn, \--okta-aws-tile, and \--mfa as parameters on the command line.  
+2) Environment variables -- You can store values in your system\'s environment variables. It overrides the configuration file.  
+3) User configuration file -- The user configuration file is updated when you run the command tokendito \--configure. tokendito uses [platformdirs](https://github.com/platformdirs/platformdirs) to store user configuration in the [tokendito.ini](tokendito.ini.md) file. This file can contain the credential details for the default profile and any named profiles.   
+
+## Environment variables and user configuration table
+
+The following table lists the environment variable and user configuration entry equivalent for the given command line option.
+| Command line option | Environment variable | User configuration |
+| ------------------- | -------------------- | ------------------ |
+| --username | TOKENDITO_OKTA_USERNAME        | okta_username | 
+| --password | TOKENDITO_OKTA_PASSWORD |   | 
+| --profile  | TOKENDITO_USER_CONFIG_PROFILE | profile |
+| --config-file | TOKENDITO_USER_CONFIG_FILE | | 
+| --loglevel | TOKENDITO_USER_LOGLEVEL | loglevel | 
+| --log-output-file | TOKENDITO_USER_LOG_OUTPUT_FILE        | log_output_file | 
+| --aws-config-file | TOKENDITO_AWS_CONFIG_FILE        | aws_config_file | 
+| --aws-output | TOKENDITO_AWS_OUTPUT        | aws_output | 
+| --aws-profile | TOKENDITO_AWS_PROFILE        | aws_profile | 
+| --aws-region | TOKENDITO_AWS_REGION       | aws_region | 
+| --aws-role-arn | TOKENDITO_AWS_ROLE_ARN       | aws_role_arn | 
+| --aws-shared-credentials-file | TOKENDITO_AWS_SHARED_CREDENTIALS_FILE        | aws_shared_credentials_file | 
+| --okta-org | TOKENDITO_OKTA_ORG        | okta_org | 
+| --okta-tile | TOKENDITO_OKTA_TILE        | okta_tile | 
+| --okta-mfa | TOKENDITO_OKTA_MFA        | okta_mfa | 
+| --okta-mfa-response | TOKENDITO_OKTA_MFA-RESPONSE        | okta_mfa_response | 
+| --quiet | TOKENDITO_USER_QUIET        | quiet | 
+
+# AWS Roles Discovery
+tokendito will discover all your available AWS Roles configured in Okta, returning a list for you to select from, simply by calling:
+```tokendito --okta-org ${YOUR ORG OKTA URL}```
```

Comment:
Here, I think we should put in an example for instance `https://acme.okta.com/` which we use throughout the code.